src/Symfony/Component/Console/Command/AlarmableCommandInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * Interface for command that listens to SIGALRM signals.
+ */
+interface AlarmableCommandInterface
+{
+    /**
+     * The method will be called before the command is run and subsequently on each SIGALRM signal.
+     *
+     * @return int The alarm time in seconds
+     */
+    public function getAlarmTime(InputInterface $input): int;
```

Comment:
I'm wondering if `time` is the right term, isn't this more an interval?

Reply:
God point, I've changed it to `getAlarmInterval()`